sceptre/template.py

```diff
@@ -234,6 +234,19 @@ def _bucket_region(self, bucket_name):
     def _domain_from_region(region):
         return "com.cn" if region.startswith("cn-") else "com"
 
+    @staticmethod
+    def _iterate_entry_points(group, name):
+        """
+        Helper to determine whether to use pkg_resources or importlib.metadata.
+        https://docs.python.org/3/library/importlib.metadata.html
+        """
+        if sys.version_info < (3, 10):
```

Comment:
I'm wondering if we should use importlib for 3.8 and above rather than just 3.10, since importlib was added in 3.8. Also, @tarkatronic just found that importlib was backported for earlier versions. We might be able to dispense with the pkg_resources import altogether if we utilized that.

Reply:
I'm not opposed to your suggestions however I think it's more risky.  I thought it might be safer going this route because we've been running and testing with `pkg_resources` for py37, py38, and py39 for a while now and it's been working.   The idea with changing it for py310 is that it's required and less risky.

Reply:
I don't think it's inherently any more risky to make this change, given that you've already made this a discrete change, separate from the Python 3.10 support.

This would just become a more holistic change-over from `pkg_resources` to `importlib`.

Reply:
I honestly don't feel all that strongly either way. I see your point about necessity... though if it works for 3.10, I don't see why it wouldn't work for 3.8+. Once 3.7 hits end of life next year, there won't really be a reason to use pkg_resources in the first place.

Reply:
I tried changing it to use importlib for py38 and higher however I could not get the tests to pass.  It's a bit confusing because the entry_points function seems to work differently in py38 & py39 than in py310.  The other odd thing is that the entry_point function for py38 & py39 does not take any parameters and I can't seem to figure out how to set group and name properties for that class.

This is what I tried..

```
        if sys.version_info < (3, 8):
            from pkg_resources import iter_entry_points
            return iter_entry_points(group, name)
        elif sys.version_info > (3, 10):
            from importlib.metadata import entry_points
            return entry_points(group=group)
        else:
            from importlib.metadata import entry_points
            return entry_points(group, name)
```

however I got this error for py39 ..
```
        else:
            from importlib.metadata import entry_points
>           return entry_points(group, name)
E           TypeError: entry_points() takes 0 positional arguments but 2 were given
```

Reply:
huh. Ok. Yeah, if it doesn't work, it doesn't work. Carry on!

Reply:
ok, i've already reverted to 5e08bf0 which was the working version